WordPress/Classes/ViewRelated/Post/Categories/PostCategoriesViewController.swift

```diff
@@ -0,0 +1,281 @@
+            let indentationLevel = indentationLevelForCategory(parentID: category.parentID, categoryCollection: categoryDict)
+            categoryIndentationDict[category.categoryID .stringValue] = NSNumber(integerLiteral: indentationLevel)
+        }
+
+        // Remove any previously selected category objects that are no longer available.
+        guard selectedCategories != nil else {
+            return
+        }
+
+        var didUpdateSelectedCategories = false
+        let updatedSelectedCategories = selectedCategories
+        self.selectedCategories = updatedSelectedCategories?.filter { category in
+            if let sortedCategories = blog.sortedCategories() as? [PostCategory], sortedCategories.contains(category), !category.isDeleted {
+                return true
+            } else {
+                didUpdateSelectedCategories = true
+                return false
+            }
+        }
+
+        // Notify the delegate of any changes for selectedCategories.
+        if didUpdateSelectedCategories {
+            delegate?.postCategoriesViewController?(self, didUpdateSelectedCategories: NSSet(array: selectedCategories!))
+        }
+
+        tableView.reloadData()
+    }
+
+    private func indentationLevelForCategory(parentID: NSNumber, categoryCollection: [NSNumber: PostCategory]) -> Int {
+        if parentID.intValue == 0 {
+            return 0
+        }
+
+        if let category = categoryCollection[parentID] {
+            return indentationLevelForCategory(parentID: category.parentID, categoryCollection: categoryCollection) + 1
+        } else {
+            return 0
+        }
+    }
+
+    //tableView
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        var result = categories.count
+        if selectionMode == .parent {
+            result = result + 1
+        }
+        return result
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.categoryCellIdentifier, for: indexPath) as? WPTableViewCell else {
+            return UITableViewCell()
+        }
+
+        var row = indexPath.row
+
+        // When showing this VC in mode CategoriesSelectionModeParent, we want the first item to be
+        // "No Category" and come up in red, to allow the user to select no category at all.
+        if selectionMode == .parent {
+            if row == 0 {
+                WPStyleGuide.configureTableViewDestructiveActionCell(cell)
+                cell.textLabel?.textAlignment = .natural
+                cell.textLabel?.text = NSLocalizedString("No Category", comment: "Text shown (to select no-category) in the parent-category-selection screen when creating a new category.")
+                if selectedCategories == nil {
```

Comment:
Could be a little more concise with

```
cell.accessoryType = selectedCategories.isEmpty ? .checkmark : .none
```

and same for the similar accessory code below.

Reply:
Done in de7c8e6af0dcdf682e2541dea294bc2694e20968